server/src/test/java/org/opensearch/search/pipeline/SearchPipelineServiceTests.java

```diff
@@ -235,34 +238,61 @@ public String getTag() {
         public String getDescription() {
             return description;
         }
+
+        @Override
+        public boolean getIgnoreFailure() {
+            return ignoreFailure;
+        }
     }
 
     private static class FakeRequestProcessor extends FakeProcessor implements SearchRequestProcessor {
         private final Consumer<SearchRequest> executor;
 
-        public FakeRequestProcessor(String type, String tag, String description, Consumer<SearchRequest> executor) {
-            super(type, tag, description);
+        public FakeRequestProcessor(String type, String tag, String description, boolean ignoreFailure, Consumer<SearchRequest> executor) {
+            super(type, tag, description, ignoreFailure);
             this.executor = executor;
         }
 
         @Override
         public SearchRequest processRequest(SearchRequest request) throws Exception {
-            executor.accept(request);
+            try {
+                executor.accept(request);
+            } catch (Exception e) {
+                if (this.getIgnoreFailure()) {
+                    return request;
+                } else {
+                    throw e;
+                }
+            }
             return request;
         }
     }
 
     private static class FakeResponseProcessor extends FakeProcessor implements SearchResponseProcessor {
         private final Consumer<SearchResponse> executor;
 
-        public FakeResponseProcessor(String type, String tag, String description, Consumer<SearchResponse> executor) {
-            super(type, tag, description);
+        public FakeResponseProcessor(
+            String type,
+            String tag,
+            String description,
+            boolean ignoreFailure,
+            Consumer<SearchResponse> executor
+        ) {
+            super(type, tag, description, ignoreFailure);
             this.executor = executor;
         }
 
         @Override
         public SearchResponse processResponse(SearchRequest request, SearchResponse response) throws Exception {
-            executor.accept(response);
+            try {
+                executor.accept(response);
+            } catch (Exception e) {
+                if (this.getIgnoreFailure()) {
+                    return response;
+                } else {
+                    throw e;
+                }
+            }
```

Comment:
Same issue as in FakeRequestProcessor -- the code in `Pipeline` should handle and ignore the exception, not the pipeline itself.

Reply:
remove catch